### PR TITLE
SE-2897: Add notebook showing portfolios with tax lot management methods

### PR DESCRIPTION
**Pull request by liberty-askew**
# Pull Request Checklist

- [x] Changes follow the [style guide](https://github.com/finbourne/sample-notebooks/blob/master/docs/FINBOURNE%20notebook%20style%20guide.ipynb)
- [x] Raised the PR against the `develop` branch
- [x] Tests pass

# Description of the PR
Add notebook showing  portfolios with different tax lot management methods examples.
Describe the code changes for the reviewers, explain the solution you have provided and how it fixes the issue

### CODE REVIEW

File: examples/use-cases/ibor/Creating portfolios with different taxlot management methods.ipynb [hunk 1]
```diff
@@ -0,0 +1,716 @@
+{
```
Comment (Tomwp) [inline examples/use-cases/ibor/Creating portfolios with different taxlot management methods.ipynb:1]: <p>I would replace 'taxlot' with 'tax lot'</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/646/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (Tomwp) [inline examples/use-cases/ibor/Creating portfolios with different taxlot management methods.ipynb:1]: <ul><li>Would change 'can be created to use any' to 'can be created using any'.</li><li>Put a space between 'taxlot'</li><li>Would use this sentence instead: 'These will determine how tax lots are used to update the cost basis of a transaction portfolio when booking various transactions.'</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/646/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (Tomwp) [inline examples/use-cases/ibor/Creating portfolios with different taxlot management methods.ipynb:1]: <p>Can you remove the column with LUIDs.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/646/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (Tomwp) [inline examples/use-cases/ibor/Creating portfolios with different taxlot management methods.ipynb:1]: <p>I don't think you need a subsection heading here. I would start 3.1 at Average Cost.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/646/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (Tomwp) [inline examples/use-cases/ibor/Creating portfolios with different taxlot management methods.ipynb:1]: <p>Might want to see if it's possible to add a tax lot ID or perhaps transaction price.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/646/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (Tomwp) [inline examples/use-cases/ibor/Creating portfolios with different taxlot management methods.ipynb:1]: <p>'brought' should be 'bought'</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/646/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (liberty-askew) [reply examples/use-cases/ibor/Creating portfolios with different taxlot management methods.ipynb:1]: <p>have done - do we want to hide this just because it's not relevant here?</p>

Comment (Tomwp) [reply examples/use-cases/ibor/Creating portfolios with different taxlot management methods.ipynb:1]: Not really relevant, but mostly because these will differ across domains.

Comment (liberty-askew) [reply examples/use-cases/ibor/Creating portfolios with different taxlot management methods.ipynb:1]: <p>Is transaction price = cost _basis? Would you like this column renaming?</p>

File: examples/use-cases/ibor/Creating portfolios with different taxlot management methods.ipynb [hunk 2]
```diff
@@ -0,0 +1,728 @@
...
```
Comment (Tomwp) [inline examples/use-cases/ibor/Creating portfolios with different taxlot management methods.ipynb:1]: <p>Should capitalize 'management' and make this consistent on all titles. Also make 'Taxlot' 'Tax lot'.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/646/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (Tomwp) [inline examples/use-cases/ibor/Creating portfolios with different taxlot management methods.ipynb:1]: <p>Remove LUIDs.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/646/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

[the remark this answers is not in the document]
Comment (liberty-askew) [reply examples/use-cases/ibor/Creating portfolios with different taxlot management methods.ipynb]: <p>good idea</p><p> </p>

File: examples/use-cases/ibor/Creating portfolios with different taxlot management methods.ipynb (hunk 1 above)
Comment (Tomwp) [reply examples/use-cases/ibor/Creating portfolios with different taxlot management methods.ipynb]: Still quite a few example of 'brought'

File: examples/use-cases/ibor/Creating Portfolios With Different Tax Lot Management Methods.ipynb [hunk 3]
```diff
@@ -0,0 +1,733 @@
+{
```
Comment (Tomwp) [inline examples/use-cases/ibor/Creating Portfolios With Different Tax Lot Management Methods.ipynb:1]: <p>I don't think we need the output of each line. Maybe just a print("All transactions upserted!") at the end.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/646/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (Tomwp) [inline examples/use-cases/ibor/Creating Portfolios With Different Tax Lot Management Methods.ipynb:1]: <ul><li>We should increase the units by a factor of 1,000. This needs to be reflected in all subsequent calculation commentary.</li><li>I would probably use a generic equity rather than AAPL as the prices aren't realistic.</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/646/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (Tomwp) [inline examples/use-cases/ibor/Creating Portfolios With Different Tax Lot Management Methods.ipynb:1]: <p>I would rephrase this;</p><p> </p><p>"Holdings before our sell transaction show the three distinct tax lots and corresponding transactions. We can see that each tax lot has a distinct cost basis based on the transaction price and total consideration of each transaction"</p><p> </p><p>I would also note mentioning that we're calling get_holdings() with the by_taxlots flag set to true.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/646/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (Tomwp) [inline examples/use-cases/ibor/Creating Portfolios With Different Tax Lot Management Methods.ipynb:1]: <ul><li>3,000 divisor should be 3,000,000</li><li>Second formula should then be 11 * 1,500,000</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/646/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (Tomwp) [inline examples/use-cases/ibor/Creating Portfolios With Different Tax Lot Management Methods.ipynb:1]: <ul><li>Would rephrase this as 'Using FIFO, our first tax lot (all 1000 units coming from txn_001) will be fully liquidated, while half of our second tax lot (500 units coming from txn_002) will be liquidated.</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/646/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (Tomwp) [inline examples/use-cases/ibor/Creating Portfolios With Different Tax Lot Management Methods.ipynb:1]: <p>Would rephrase this using similar language to FIFO proposed changes.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/646/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

Comment (Tomwp) [inline examples/use-cases/ibor/Creating Portfolios With Different Tax Lot Management Methods.ipynb:1]: <p>In your second sentence, I would replace 'The equities from <code>txn_001'</code></p><p>with 'All units from...'. Same for Lowest first.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/646/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='27'/>

Comment (liberty-askew) [reply examples/use-cases/ibor/Creating Portfolios With Different Tax Lot Management Methods.ipynb:1]: <p>Might be a bit bulky to implement as each is a separate call to the method. I've update the print out so it is less repetitive.</p><p> </p>

Comment (liberty-askew) [reply examples/use-cases/ibor/Creating Portfolios With Different Tax Lot Management Methods.ipynb:1]: <ul><li>So 1,000 -&gt; 1,000,000 units or the other way around?</li></ul>

Comment (liberty-askew) [reply examples/use-cases/ibor/Creating Portfolios With Different Tax Lot Management Methods.ipynb:1]: <p>Changes to inst_DEMO</p><p> </p>

File: examples/use-cases/ibor/Creating Portfolios With Different Tax Lot Management Methods.ipynb [hunk 4]
```diff
@@ -0,0 +1,847 @@
...
```
Comment (Tomwp) [inline examples/use-cases/ibor/Creating Portfolios With Different Tax Lot Management Methods.ipynb:1]: <p>Can remove the two commas.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/646/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (Tomwp) [inline examples/use-cases/ibor/Creating Portfolios With Different Tax Lot Management Methods.ipynb:1]: <p>I would rephrase the first sentence as follows:</p><p> </p><p><strong><em>We'll now look at the final positions of our portfolios under the 5 accounting methods described above following a sale of 1,500,000 units in a single sell transaction. </em></strong></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/646/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

**Review by Tomwp — APPROVED** [review]
Approved
